### DISCUSSION TITLE
using new spec parser and astToDOM in combination with parameters and selection

### DISCUSSION THREAD

**Discussion by ricksmit3000** [uwdata/mosaic#324] [Q&A]
With the old spec parser I was able to do something like this:
```ts
const globalSelection = vg.Selection.intersect();

vg.parseSpec(spec, {
  params: [["global", globalSelection]],
}).then((plot) => {
   containerRef.current?.replaceChildren(plot);
})
```

With the new spec parser I thought I had to change it to something like this:

```ts
const ast = parseSpec(spec);

const globalSelection = vg.Selection.intersect();
const activeParams = new Map();
activeParams.set("global", globalSelection);

astToDOM(ast, { api: vg, activeParams }).then(({ element, params }) => {
  containerRef.current?.replaceChildren(element);
});
```
The plot renders, but it doesn't seem to apply my selection - is it obvious what I'm doing wrong?

I can also make a small reproducible example if that'd help.

**Comment by jheer** [comment]
This looks like a bug. I've opened an issue to fix this. Thanks!

**Comment by ricksmit3000** [comment]
I see you already fixed it - amazing, thank you!
